### PR TITLE
feat(Tests): extra commands to run unit tests TDD style as well as e2e in interactive mode. Also improved README.

### PR DESCRIPTION
**Pull request by NathanWalker**
This adds the following:

- extra commands to make testing more convenient and flexible:
  - For unit tests: `npm run test-watch` for TDD style development
  - For e2e tests: `npm run e2e-live` to place protractor in interactive mode (details mentioned in README)

Also improved the README to remove all broken links and some cleanup.
Various other small refactorings to help improve and strengthen.

### CODE REVIEW

File: src/app/components/home/home.ts [hunk 1]
```diff
@@ -9,13 +9,13 @@ import {Http} from 'angular2/http';
     styles: [require('./home.css')],
     template: require('./home.html')
 })
-export class Home {
+export class Home implements OnInit {
```
Comment (NathanWalker) [inline src/app/components/home/home.ts]: This is recommended as discussed here: https://github.com/angular/angular/issues/5814

**Comment by ocombe** [comment]
Awesome, thanks.
You don't like the pink ? :D

**Comment by NathanWalker** [comment]
I'm actually a big fan of pink, but it was a tad bit hard to read on that
gray background. Yellow may pop better. But if it's not any better to you,
by all means bring the pink back :)

**Comment by ocombe** [comment]
No worries, I actually started typing #F and took whatever webstorm offered, I was just testing my build at the time :)

**Comment by ocombe** [comment]
I improved the design a lot in one of my lasts commits :-)

**Pull request merged**